### PR TITLE
Feature/dev 12829 createing shared logging library

### PR DESCRIPTION
**Pull request by simon-technocloud**
Basic unit tests have been added

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import VmdHapiLogger from './logging/vmd-logger'
+
+export {
+    VmdHapiLogger
+}
```
Comment (mogusbi-motech) [inline index.js:5]: This file could be made to be more concise;

`export { default as VmdHapiLogger } from './logging/vmd-logger'`

File: logging/helper.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            level: logLevel,
+            logDirectory,
+            fileNamePattern: serviceName + '-<DATE>.log',
+            dateFormat: 'YYYY.MM.DD'
+        }
+
+        const rollingFileLogger = simpleNodeLogger.createRollingFileLogger(rollingFileLoggerOptions)
+        scheduleLogDeletion(rollingFileLogger, logDirectory, serviceName, nameOffset)
+        return rollingFileLogger
+    } else {
```
Comment (mogusbi-motech) [inline logging/helper.js:25]: You should avoid else returns: https://eslint.org/docs/rules/no-else-return

File: logging/helper.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+}
+
+function buildLoggingUrl (httpMethod, url) {
+    return `[${httpMethod.toUpperCase()}] ${url}`
+}
+
+export {
+    buildLogger,
+    buildLogMessage
+}
```
Comment (mogusbi-motech) [inline logging/helper.js:109]: This is opinionated but you can avoid this export statement at the bottom of the file by just adding the word `export` before the function names i.e. `export function buildLogger` and `export function buildLogMessage`

File: logging/vmd-logger.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    constructor (serviceName, config, logDirectory = 'logs') {
+        if (!serviceName) {
+            throw new Error('HapiLogger requires a service name')
+        }
+
+        const isFileLogging = config.requiresFileLogging || false
+        const logLevel = config.logLevel || 'info'
+        this.logger = buildLogger(serviceName, isFileLogging, logDirectory, logLevel)
+
+        return this.logger
```
Comment (mogusbi-motech) [inline logging/vmd-logger.js:14]: Should a constructor be returning?

**Review by mogusbi-motech — COMMENTED** [review]
Just a few comments/questions. As an aside, do you need unit tests?

**Review by PBarnard — APPROVED** [review]
Nice 👍

**Review by meyumeapps — APPROVED** [review]
LGTM!

File: logging/vmd-logger.js (hunk 4 above)
Comment (simon-technocloud) [reply logging/vmd-logger.js:14]: Good point...

File: logging/helper.js (hunk 3 above)
Comment (simon-technocloud) [reply logging/helper.js:109]: Ok, cool do we have a standard or project preference?

File: logging/helper.js (hunk 2 above)
Comment (simon-technocloud) [reply logging/helper.js]: Agreed...I copied that code, but will address this!

File: logging/helper.js (hunk 3 above)
Comment (mogusbi-motech) [reply logging/helper.js]: I guess this is really down to WW. All I would say is that you're already directly exporting classes so why are functions any different?

**Review by mogusbi-motech — APPROVED** [review]
LGTM

**Review by UntappedSolutions — APPROVED** [review]
LGTM

**Pull request merged**